dsref/load.go

```diff
@@ -10,14 +10,15 @@ import (
 // ErrNoHistory indicates a resolved reference has no HEAD path
 var ErrNoHistory = fmt.Errorf("no history")
 
-// Loader loads and opens a dataset. The only useful implementation of the
-// loader interface is in github.com/qri-io/qri/lib.
-// TODO(b5) - This interface is a work-in-progress
+// ErrRefNotResolved is the error when LoadResolved is called on a non-resolved ref
+var ErrRefNotResolved = fmt.Errorf("ref has not been resolved, cannot load")
+
+// Loader loads datasets
 type Loader interface {
-	LoadDataset(ctx context.Context, ref Ref, source string) (*dataset.Dataset, error)
+	// LoadDataset will parse the ref string, resolve it, and load the dataset
+	// from whatever store contains it
+	LoadDataset(ctx context.Context, refstr string) (*dataset.Dataset, error)
+	// LoadResolved loads a dataset that has already been resolved by finding it
+	// at the given location. Will fail if the Ref has no Path with ErrRefNotResolved
```

Comment:
I'd prefer dsref get a new `Resolved() bool` or `IsResolved() bool` method, and use that as the check. The formal definition of resolution is having an InitID. It's possible to have a resolved dataset with no path (empty history). Having that logic on dsref will avoid ad-hoc checks across the codebase like this

Reply:
I also like that better, making it explicit is good.